articles/openstack/ostack-how-basic-heat-template.md

```diff
@@ -0,0 +1,164 @@
+
+## Overview
+
+This is a basic heat orchestration template create a heat stack.
+
+This template will create the following and connect them up.
+- A network
+- A static IP
+- A router
+- An instance
+- A floating IP
+- A volume
+
+## The heat template
+
+> [!NOTE]
+> Make sure you check the indentation and file extension. It needs to be in yaml format.
+
+```yaml
+heat_template_version: 2016-10-14
+
+parameters:
+  flavor:
+    type: string
+    description: I am using the smallest flavor available because i'll be spinning up a cirros instance. You can use an environment file to override the defaults.
+    default: t1.tiny
+    constraints:
+      - custom_constraint: nova.flavor
+  image:
+    type: string
+    description: This uses a cirros image but you can create an environment file to change the default values.
+    default: cirros
+    constraints:
+      - custom_constraint: glance.image
+  heat_volume_size:
+    type: number
+    label: Volume Size (GB)
+    description: External Volume Size in GB
+    default: 1
+
+resources:
+  heat_volume:
+    type: OS::Cinder::Volume
+    properties:
+      size: { get_param: heat_volume_size }
+  heat_volume_attachement:
+    type: OS::Cinder::VolumeAttachment
+    properties:
+      volume_id: { get_resource: heat_volume }
+      instance_uuid: { get_resource: heat_server }
+  heat_network:
+    type: OS::Neutron::Net
+    properties:
+      admin_state_up: true
+      name: heat_network
+  heat_network_subnet:
+    type: OS::Neutron::Subnet
+    properties:
+      network: { get_resource: heat_network }
+      cidr: "10.1.1.0/24"
+      dns_nameservers: ["8.8.8.8"]
+      gateway_ip: "10.1.1.1"
+      ip_version: 4
+  heat_router:
+    type: OS::Neutron::Router
+    properties:
+      external_gateway_info: { network: internet }
+      name: heat_router
+  heat_router_interface:
+    type: OS::Neutron::RouterInterface
+    properties:
+      router_id: { get_resource: heat_router }
+      subnet: { get_resource: heat_network_subnet }
+  heat_server_port:
+    type: OS::Neutron::Port
+    properties:
+      network: { get_resource: heat_network }
+      fixed_ips:
+        - subnet_id: { get_resource: heat_network_subnet }
+  heat_server:
+    type: OS::Nova::Server
+    properties:
+      name: heat_server
+      flavor: { get_param: flavor }
+      image: { get_param: image }
+      networks:
+        - port: { get_resource: heat_server_port}
+  heat_server_public_ip:
+    type: OS::Neutron::FloatingIP
+    properties:
+      floating_network: "internet"
+  heat_server_ip_assoc:
+    type: OS::Neutron::FloatingIPAssociation
+    properties:
+      floatingip_id: { get_resource: heat_server_public_ip }
+      port_id: { get_resource: heat_server_port }
+
+outputs:
+  heat_server_public_ip:
+    description: IP Address of the deployed heat_server instance
+    value: { get_attr: [ heat_server_public_ip, floating_ip_address ]}   
+```
+
+## Deploying the stack
+
+> [!NOTE]
```

Comment:
I'd suggesting adding some introductory text here. For example:

After you create the template, you must deploy it to <whatever> so that <whatever>.